examples/src/test/java/io/numaproj/numaflow/examples/map/evenodd/EvenOddFunctionTest.java

```diff
@@ -0,0 +1,68 @@
+package io.numaproj.numaflow.examples.map.evenodd;
+
+
+import io.numaproj.numaflow.mapper.Datum;
+import io.numaproj.numaflow.mapper.Message;
+import io.numaproj.numaflow.mapper.MessageList;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.Test;
+
+import java.util.Iterator;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@Slf4j
+public class EvenOddFunctionTest {
+
+    @Test
+    public void testEvenNumber() {
+        Datum datum = mock(Datum.class);
+        when(datum.getValue()).thenReturn("2".getBytes());
+
+        EvenOddFunction evenOddFunction = new EvenOddFunction();
+        MessageList result = evenOddFunction.processMessage(new String[]{}, datum);
+
+        Iterator<Message> iterator = result.getMessages().iterator();
+        Message message = iterator.hasNext() ? iterator.next() : null;
+
+        // The message should have the key "even" and tag "even-tag"
+        assert message != null;
```

Comment:
This shouldn't be needed in every single test. Ideally we don't need to worry about the message being null and can just assert of the data.